### PR TITLE
Pattern Library: Ensure that form submission disabled

### PR DESCRIPTION
**Pull request by olaseni**
Fixes issue: Automattic/dotcom-forge#6584

We add an event handler to forms with patterns to ensure that forms cannot be submitted.

Testing Instructions
---

1. Navigate to https://wordpress.com/patterns/contact#pattern-13979
3. Click on the `Contact us` heading in the pattern
4. Click <kbd>Tab</kbd> on your keyboard
5. The first input field should receive focus. Enter some text
6. Click <kbd>Tab</kbd> 
7. The second input field should receive focus. Enter some text
8. Click <kbd>Enter</kbd> 
9. Confirm that nothing happens, or you are not able to submit said form

### CODE REVIEW

**Comment by olaseni** [comment]
This is a bit different than at dotorg where it's not handled it seems: https://wordpress.org/patterns/pattern/footer-with-search-site-title-and-credit-line/.

File: client/my-sites/patterns/components/pattern-preview/index.tsx [hunk 1]
```diff
@@ -271,6 +271,29 @@ function PatternPreviewFragment( {
...
+
+		if ( ! iframe?.contentDocument ) {
+			return;
+		}
+
+		const forms = Array.from( iframe.contentDocument.querySelectorAll( 'form' ) );
+
+		if ( ! forms.length ) {
+			return;
+		}
```
Comment (fredrikekelund) [inline client/my-sites/patterns/components/pattern-preview/index.tsx]:
```suggestion
		const forms = iframe.contentDocument.querySelectorAll( 'form' );
```
Somewhat nitpicky, but we don't actually need to cast the `NodeList` to an array (`forEach` is supported by `NodeList`, too). We also don't really need to return early if the list is empty.

File: client/my-sites/patterns/components/pattern-preview/index.tsx [hunk 2]
```diff
@@ -271,6 +271,29 @@ function PatternPreviewFragment( {
...
+
+		const onFormSubmit = ( event: { preventDefault: () => void } ) => event.preventDefault();
```
Comment (fredrikekelund) [inline client/my-sites/patterns/components/pattern-preview/index.tsx]:
```suggestion
		const onFormSubmit = ( event: SubmitEvent ) => event.preventDefault();
```
`SubmitEvent` is the type TS infers for `submit` events.

**Review by fredrikekelund — APPROVED** [review]
Two small comments, but other than that, this looks good and works as expected 👍

File: client/my-sites/patterns/components/pattern-preview/index.tsx (hunk 1 above)
Comment (olaseni) [reply client/my-sites/patterns/components/pattern-preview/index.tsx]: > We also don't really need to return early if the list is empty.

The advantage of returning early is, we don't need to create nor allocate unneeded resources.  Also even though the list is empty, a function is invoked, nevertheless.

**Pull request merged**